File: avaliacao-server/src/main/java/br/edu/ufca/avaliacao/model/Resposta.java
package br.edu.ufca.avaliacao.model;

import lombok.Data;
import lombok.EqualsAndHashCode;

import javax.persistence.*;
import java.math.BigDecimal;

@Data
@Entity
@EqualsAndHashCode(of = "id")
public class Resposta implements BaseEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @OneToOne
    @JoinColumn(name = "ciclo_id")
    private Pergunta pergunta;

    @ManyToOne
    @JoinColumn(name = "avaliacao_id")
    private Avaliacao avaliacao;

    private BigDecimal valor;

    @Data
    public static class Import {
        private BigDecimal resposta;
    }

}

File: avaliacao-client/src/app/core/model/resposta.ts
import {BaseEntity} from './base-entity';
import {Pergunta} from './pergunta';
import {Avaliacao} from './avaliacao';

export class Resposta implements BaseEntity {

  id: number;
  pergunta: Pergunta;
  avaliacao: Avaliacao;
  valor: number;

}
